master/internal/resourcemanagers/config.go

```diff
@@ -26,12 +26,9 @@ func (r *ResourceConfig) ResolveResource() error {
 		r.ResourceManager.AgentRM = &AgentResourceManagerConfig{}
 	}
 	if r.ResourceManager.AgentRM != nil && r.ResourcePools == nil {
-		r.ResourcePools = []ResourcePoolConfig{
-			{
-				PoolName:                 defaultResourcePoolName,
-				MaxCPUContainersPerAgent: 100,
-			},
-		}
+		defaultPool := *defaultRPConfig()
+		defaultPool.PoolName = defaultResourcePoolName
+		r.ResourcePools = []ResourcePoolConfig{defaultPool}
```

Comment:
nit but seems either `defaultRPConfig` should not return a pointer or `r.ResourcePools` should be a list pointer.

Reply:
I vote no pointer.  I think that our config handling is *very* loose about what things should be pointers vs what things are pointers because something similar to that was a pointer somewhere else.